### PR TITLE
SY CHOU, CW LIN modified. Add TransactionGraph, TransactionGraphQueue…

### PR DESCRIPTION
**Pull request by tom1236868**
…, SyncRequest, TimeSync classes. Add new functionalities, including depedency graph, feature collection, transmission time synchronization. Wait for merging

### CODE REVIEW

**Comment by FrankCCCCC** [comment]
As for complete commit message, here it is:

SY CHOU, CW LIN modified. Add TransactionGraph, TransactionGraphQueue, SyncRequest, TimeSync classes. Add new functionalities, including depedency graph, feature collection, transmission time synchronization. Wait for merging

**Comment by FrankCCCCC** [comment]
image

Note that, in the line 76 ~ 80 of TPartStoredProcedureTask.java, since we cannot find the definition of "stopComponentTimer("Generate plan", sinkStopTime)", the compiler will show error for the code. In order to export the jar file, we don't enable that part of code temporarily.

File: src/main/java/org/elasql/cache/RemoteRecordReceiver.java [hunk 1]
```diff
@@ -20,5 +20,5 @@
 public interface RemoteRecordReceiver {
 	
 	void cacheRemoteRecord(Tuple t);
-	
 }
+
```
Comment (xuan19970825) [inline src/main/java/org/elasql/cache/RemoteRecordReceiver.java:24]: This file is not related. Please undo the changes.

**Review by xuan19970825 — CHANGES_REQUESTED** [review]
There are some files that are not related to the change. We hope you can remove these files from the commit.
`src/main/java/org/elasql/cache/RemoteRecordReceiver.java`
`src/main/java/org/elasql/storage/tx/concurrency/ConservativeOrderedCcMgr.java`
`src/main/java/org/elasql/schedule/tpart/TPartScheduler.java`

Some codes are commented in this file, could you please tell us the reason?
`src/main/java/org/elasql/storage/tx/concurrency/ConservativeOrderedLockTable.java`

**Review by SLMT — APPROVED** [review]
Great! Thanks for your contribution. I will do some clean up after merging.

**Review by xuan19970825 — APPROVED** [review]

**Pull request merged**